src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/InternalCpxSvDiscoverFromLongReadsSpark.java

```diff
@@ -0,0 +1,228 @@
+            = new StructuralVariationDiscoveryArgumentCollection.DiscoverVariantsFromContigsAlignmentsSparkArgumentCollection();
+
+    @Argument(doc = "file containing non-canonical contig names (e.g chrUn_KI270588v1) in the reference, human reference assumed when omitted",
+            shortName = "nonCanoTigFile",
+            fullName = "nonCanonicalContigNamesFile", optional = true)
+    public String nonCanonicalContigNamesFile;
+
+    @Argument(doc = "output directory", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+              fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME)
+    private String outputDir;
+
+    @Argument(doc = "output SAM files", shortName = "wSAM",
+              fullName = "writeSAM", optional = true)
+    private boolean writeSAMFiles = false;
+
+    @Override
+    public boolean requiresReference() {
+        return true;
+    }
+
+    @Override
+    public boolean requiresReads() {
+        return true;
+    }
+
+    @Override
+    public ReadFilter makeReadFilter() {
+        return ReadFilterLibrary.MAPPED;
+    }
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+
+        final JavaRDD<GATKRead> reads = getReads();
+        final SAMFileHeader header = getHeaderForReads();
+
+        // filter alignments and split the gaps
+        final JavaRDD<AlignedContig> contigsWithAlignmentsReconstructed =
+                InternalFilterLongReadAlignmentsSAMSpark.filterByScore(reads, header, nonCanonicalContigNamesFile, localLogger)
+                        .filter(lr -> lr.alignmentIntervals.size()>1).cache();
+
+        // split up the long reads by their possible type of SV
+        final EnumMap<RawTypes, JavaRDD<AlignedContig>> splitUpLongReads =
+                splitReadsByPossiblyRawTypes(contigsWithAlignmentsReconstructed, localLogger);
+
+        if ( !FileUtils.createDirInBucketToWriteTo(outputDir) )
+            throw new GATKException("Could not create directory " + outputDir + " to write results to.");
+
+        if (writeSAMFiles) {
+            final Broadcast<SAMFileHeader> headerBroadcast = ctx.broadcast(header);
+            splitUpLongReads.forEach((k, v) -> writeSAM(v, k.name(), reads, headerBroadcast, outputDir, localLogger));
+        }
+
+        new ForSimpleInsDel()
+                .inferSvAndWriteVCF(splitUpLongReads.get(RawTypes.InsDel), outputDir+"/"+RawTypes.InsDel.name()+".vcf",
+                        ctx.broadcast(getReference()), discoverStageArgs.fastaReference, getAuthenticatedGCSOptions(),
+                        localLogger);
+
+//        new ForSimpleStrandSwitch()
+//                .inferSvAndWriteVCF(splitUpLongReads.get(RawTypes.Inv), outputDir+"/"+RawTypes.Inv.name()+".vcf",
+//                        ctx.broadcast(getReference()), discoverStageArgs.fastaReference, getAuthenticatedGCSOptions(),
+//                        localLogger);
+    }
+
+    private enum RawTypes {
+        MultiConf, Inv, InsDel, SegDupOrMEI, Cpx;
```

Comment:
`MultiConf` doesn't really make it obvious what these things are. Might `Ambiguous` be a better name?